pkg/services/accesscontrol/database/database_mig.go

```diff
@@ -14,14 +14,14 @@ func AddAccessControlMigrations(mg *migrator.Migrator) {
 			{Name: "updated", Type: migrator.DB_DateTime, Nullable: false},
 		},
 		Indices: []*migrator.Index{
-			{Cols: []string{"role_id"}},
+			{Cols: []string{"role_id", "permission", "scope"}, Type: migrator.UniqueIndex},
```

Comment:
Shouldn't we add index instead of replacing? When we performing search for permissions, we're relying on `role_id`, but not filtering by permission and scope, so we need `role_id` indexed as well.

Reply:
```suggestion
          {Cols: []string{"role_id"}},
          {Cols: []string{"role_id", "permission", "scope"}, Type: migrator.UniqueIndex},
```

Reply:
Composite index in [MySQL](https://dev.mysql.com/doc/refman/8.0/en/multiple-column-indexes.html) and [PostgreSQL](https://www.postgresql.org/docs/9.6/indexes-multicolumn.html) would be sufficient to run queries using indexes on the 1st, 2nd, n-th subset of indexes, in other words if the query would use `role_id` only, or (`role_id` , `permission`) or (`role_id`, `permission`, `scope`), in all cases query planner will use the indexes. The same is in [SQLite](https://sqlite.org/queryplanner.html#_searching_and_sorting_with_a_multi_column_index).

That said - one thing I am not sure is if I should add a new migration or update the current one? What happens if the migration has already been run? 🤔

Reply:
If the migration is already run, then it's not being executed.

Reply:
We usually add new migrations for the code that added to master (since people can use nightly builds), but in this case feature is behind the flag, so replacing migration should be relatively safe.